src/goto-programs/show_goto_functions.cpp

```diff
@@ -67,6 +68,38 @@ void show_goto_functions(
       bool is_internal=(has_prefix(id2string(it->first), CPROVER_PREFIX) ||
                         it->first==ID__start);
       json_function["isInternal"]=jsont::json_boolean(is_internal);
+
+      if(it->second.body_available())
+      {
+        json_arrayt json_instruction_array=json_arrayt();
+
+        for(const goto_programt::instructiont &instruction :
+          it->second.body.instructions)
+        {
+          json_objectt instruction_entry=json_objectt();
+
+          std::ostringstream instruction_id_builder;
+          instruction_id_builder << instruction.type;
```

Comment:
May I suggest to add some to_string or as_string method for instruction types? This won't be the only place that uses a one-off stringstream.